src/components/select/select.ts

```diff
@@ -0,0 +1,122 @@
+import {
+    Component,
+    ContentChildren,
+    EventEmitter,
+    forwardRef,
+    Input,
+    ModuleWithProviders,
+    NgModule,
+    Output,
+    QueryList,
+    ViewChild,
+    ViewEncapsulation
+} from '@angular/core';
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from '@angular/forms';
+import { BrowserModule } from '@angular/platform-browser';
+import { MdlPopoverModule, MdlPopoverComponent } from '../popover';
+import { MdlOptionComponent } from './index';
+
+function randomId() {
+    const S4 = () => (((1+Math.random())*0x10000)|0).toString(16).substring(1);
+    return (S4()+S4());
+}
+
+const MDL_SELECT_VALUE_ACCESSOR: any = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => MdlSelectComponent),
+    multi: true
+};
+
+@Component({
+    moduleId: module.id,
+    selector: 'mdl-select',
+    host: {
+        '[class.mdl-select]': 'true'
+    },
+    templateUrl: 'select.html',
+    styleUrls: ['select.scss'],
+    encapsulation: ViewEncapsulation.None,
+    providers: [MDL_SELECT_VALUE_ACCESSOR]
+})
+export class MdlSelectComponent implements ControlValueAccessor {
+    @Input() ngModel: any;
+    @Input('placheholder') placeholder: string = '';
+    @Output() private ngModelChange: EventEmitter<any> = new EventEmitter(true);
+    @ViewChild(MdlPopoverComponent) private popoverComponent: MdlPopoverComponent;
+    @ContentChildren(MdlOptionComponent) private optionComponents: QueryList<MdlOptionComponent>;
+    private textfieldId: string;
+    private text: string = '';
+    private textByValue: any = {};
+    private onChange: any = Function.prototype;
+    private onTouched: any = Function.prototype;
+
+    constructor() {
+        this.textfieldId = `mdl-textfield-${randomId()}`;
+    }
+
+    ngAfterViewInit() {
+        this.textByValue[''] = this.placeholder;
+        this.optionComponents.forEach((selectOptionComponent: MdlOptionComponent) => {
+            selectOptionComponent.onSelect = this.onSelect.bind(this);
+            this.textByValue[String(selectOptionComponent.value)] = selectOptionComponent.text;
+        });
+        this.renderValue(this.ngModel);
```

Comment:
here is the problem. this method gets called after writeValue has been called and already rendered the view. but at the time writevalue calls renderValue the options are not knwon (-> text becomes undefined). If the renderValue method is called from ngAfterViewInit the options are set and text becomes 'Bryan Cranston'. The TestBed is designed to check those unexpected changes. I did not found a quick solution to make sure that there are no changes after the change detection. please see https://github.com/angular/angular/issues/6005 as a starting point...